<?xml version="1.0"?>
<launch>

  <arg name="pcd_file_name" default="$(find mlcpp)/resource/bigben1.pcd"/>

  <!-- Rviz  -->
  <node name="visualizer" pkg="rviz" type="rviz" output="screen" args="-d $(find mlcpp)/rviz.rviz"/>


  <!-- Basic params  -->
  <param name="/debug_mode" value="false"/>
  <param name="/infile" value="$(arg pcd_file_name)"/>
  <rosparam param="/cam_intrinsic">[640, 480, 319.998, 319.998, 320.5, 240.5]</rosparam> <!-- width, height, f_x, f_y, c_x, c_y -->

  <!-- MLCPP params -->
  <param name="/slice_height" value="7.0"/> <!-- slicing target PCL every this meters -->
  <param name="/max_dist" value="16.0"/> <!-- between viewpoint and target pcl point -->
  <param name="/max_angle" value="75.0"/> <!-- between viewpoint and target pcl point's normal -->
  <param name="/view_pt_dist" value="10.0"/> <!-- from points -->
  <param name="/view_pt_each_dist" value="2.0"/> <!-- between each viewpoints -->
  <param name="/view_overlap" value="0.4"/> <!-- overlap bet two viewpoints -->
  <param name="/TSP_trial" value="500"/>

  <!-- main node  -->
  <node name="mlcpp_node" pkg="mlcpp" type="mlcpp_slice_node" output="screen"/>
  <!-- launch-prefix="gdb -ex run --> <!--args" -->


</launch>